### PR TITLE
Grafana UI: `FilterPill.story.tsx` - Replace `HorizontalGroup` with `Stack`

### PR DESCRIPTION
**Pull request by eledobleefe**
**What is this feature?**

This is part of the epic https://github.com/grafana/grafana/issues/85510

**Why do we need this feature?**
To replace deprecated layout components with new ones.

**Who is this feature for?**
Everybody

**Which issue(s) does this PR fix?:**

Fixes #

**Special notes for your reviewer:**

**Before:**
Captura de pantalla 2024-04-08 a las 15 27 25

Captura de pantalla 2024-04-08 a las 15 27 15

**After:**
Captura de pantalla 2024-04-08 a las 15 27 52

Captura de pantalla 2024-04-08 a las 15 28 11

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

File: .betterer.results [hunk 1]
```diff
@@ -1,5 +1,5 @@
 // BETTERER RESULTS V2.
-//
+//
```
Comment (JoaoSilvaGrafana) [inline .betterer.results:2]: Is this intended?

**Review by JoaoSilvaGrafana — APPROVED** [review]
Looks good just a small nit question!

File: .betterer.results (hunk 1 above)
Comment (eledobleefe) [reply .betterer.results:2]: Nope

Comment (eledobleefe) [reply .betterer.results:2]: I checked on [main](https://github.com/grafana/grafana/blob/d0312af571ba8460c5dab26282e0f7000730e4a5/.betterer.results#L2) and it seems line 2 is already there. Is it just the blank space?

Comment (eledobleefe) [reply .betterer.results]: @JoaoSilvaGrafana fixed!

**Review by JoaoSilvaGrafana — APPROVED** [review]

**Pull request merged**